riscv/insns/mret.h

```diff
@@ -7,8 +7,12 @@ if (prev_prv != PRV_M)
   s = set_field(s, MSTATUS_MPRV, 0);
 s = set_field(s, MSTATUS_MIE, get_field(s, MSTATUS_MPIE));
 s = set_field(s, MSTATUS_MPIE, 1);
-s = set_field(s, MSTATUS_MPP, PRV_U);
 s = set_field(s, MSTATUS_MPV, 0);
+if (p->extension_enabled('H')) {
+  s = set_field(s, MSTATUS_MPP, 0);
+} else {
+  s = set_field(s, MSTATUS_MPP, p->extension_enabled('U') ? PRV_U : PRV_M);
+}
```

Comment:
If `p->extension_enabled('H')` is true, then `p->extension_enabled('U')` is also necessarily true.  So you should be able to delete lines 11-13 and 15, and keep only line 14.

Reply:
agree and will be fix in next update

Reply:
done